client/components/data/domain-management/index.jsx

```diff
@@ -59,7 +59,7 @@ const DomainManagementData = createReactClass( {
 		const { selectedSite: prevSite } = this.props;
 		const { selectedSite: nextSite } = nextProps;
 
-		if ( nextSite !== prevSite ) {
+		if ( nextSite && nextSite !== prevSite ) {
 			fetchDomains( nextSite.ID );
```

Comment:
I think we can simplify things further by also fetching selectedSiteId from redux in connect. In addition to that we could maybe update `fetchDomains` to no-op when it gets a falsey value, this would make much less work for the caller.

Reply:
eg
```
if ( this.props.selectedSiteId !== nextProps.selectedSiteId ) {
    fetchDomains( nextProps.selectedSiteId );
}
```

Reply:
Note that QueryComponents like `QuerySitePlans` can handle falsey siteId input (for ease of use). So similarly we can update this to:
```jsx
<QuerySitePlans siteId={ this.props.selectedSiteId } />
```

If `QueryContactDetailsCache` should only fetch data when there's a selected site available (in general) consider moving that check to the query component itself. If it's a one off we can of course call:
```jsx
{ this.props.selectedSiteId && <QueryContactDetailsCache /> }
```